paasta_tools/marathon_tools.py

```diff
@@ -94,7 +98,75 @@
 logging.getLogger('marathon').setLevel(logging.WARNING)
 
 MarathonServers = namedtuple('MarathonServers', ['current', 'previous'])
-MarathonClients = namedtuple('MarathonClients', ['current', 'previous'])
+
+
+_RendezvousHashT = TypeVar('_RendezvousHashT')
+
+
+def rendezvous_hash(
+    choices: Sequence[_RendezvousHashT],
+    key: str,
+    salt: str='',
+    hash_func: Callable[[str], str]=get_config_hash,
+) -> _RendezvousHashT:
+    """For each choice, calculate the hash of the index of that choice combined with the key, then return the choice
+    whose corresponding hash is highest.
+
+    :param choices: A sequence of arbitrary values. The "winning" value will be returned."""
+    max_hash_value = None
+    max_hash_choice = None
+    for i, choice in enumerate(choices):
+        str_to_hash = MESOS_TASK_SPACER.join([str(i), key, salt])
```

Comment:
It seems strange to me to use the MESOS spacer here. 
Does this matter? I expected this function to be very generic and not need anything paasta-related at all

Reply:
I can make this any arbitrary format.